### ISSUE TITLE
`Dropout()` at prediction time

### ISSUE THREAD

**Issue by muammar** [chemprop/chemprop#185]
I was looking inside the code but wanted to confirm if the loaded model at prediction time is set to `.eval()` mode? I would say yes because, at prediction time, one would not want to have the `.Dropout()` function applied and having different outputs every time. Also, would it be possible to point out where this is actually happening? Is it here?

https://github.com/chemprop/chemprop/blob/93e0aefda97e7153da7715120ff55a66863948f9/chemprop/train/predict.py#L10-L26

Best,

**Comment by cjmcgill** [comment]
You have interpreted this correctly. The `model.eval()` line changes the model into eval mode with disables dropout and other tools like batch norm that are used in training but not in prediction. That eval mode is active inside the `predict` function but is not persistent after the predictions are returned. This is used when chemprop is performing predictions and when chemprop is evaluating scores on validation or test sets during training.

Not what you were asking but related: the line `with torch.no_grad()` line is also important for evaluations during training, as it temporarily stops updates to the parameter gradients that might occur with using the model so that they don't interfere with later optimizer steps during training.

**Comment by muammar** [comment]
Hey @cjmcgill, thanks for your reply; I appreciate it. I think I was a bit lazy and did not go through the whole `predict()` function. I agree with all you wrote. I am doing some experiments and probably will try to propose a PR later, but I need to be sure. 

I will proceed to close this issue. 

Best,

**Comment by cjmcgill** [comment]
What PR are you considering? It's possible that we already have the capability for what you want to do. And if not, we will be able to help in the PR process.
